Tests/streamsTests/TransformTests.swift

```diff
@@ -139,6 +140,55 @@ class NozeIOTransformTests: NozeIOTestCase {
     XCTAssertEqual(concatData!.count, fix.characters.count * 2)
   }
 
+  func testExportingOnlyReadableStreamOfDuplex() {
+    enableRunLoop() // pipe requires async
+
+    // Situation that we try to reproduce:
+    // 1. We have some Instance that produces data as a ReadStream 
+    //    (e.g. Database Connection)
+    // 2. We want to transform this data before we send it to the consumer
+    //    with a TransformStream
+    // 3. The userfacing interface shall hide the transform stream. Only the
+    //    readable end of the duplex shall be exposed.
+
+    class Connection {
+
+      var stream: ReadableStream<String>? = nil
+
+      func query() -> ReadableStream<String> {
+        self.stream = Readable<String>()
+        let handOver = Transform<String, String>() { (items, _, done) in
+          done(nil, items)
+        }
+        return ( self.stream! | handOver ).readStream
```

Comment:
Doesn't make sense to me. The 'readStream' of a duplex is an implementation detail and should be completely private. The Duplex stream itself *is* a read stream (`DuplexStream<ReadType, WriteType> : Stream, **GReadableStreamType**`). Not sure why this is marked as `public`, I think there have been issues with the Linux linker (though I usually tagged such issues with a marker comment).